### PR TITLE
consensus, rpc: Kermit's mom hardfork (2671700)

### PR DESCRIPTION
**Pull request by jamescowens**
This sets the mainnet block v12, poll v3, and project v2 height to 2671700 (i.e. all coincident on the same height). This height will occur on or about Sunday, Aug 31, 2022. This also changes the masterkeys and alertkeys.

### CODE REVIEW

File: src/rpc/net.cpp [hunk 1]
```diff
@@ -384,6 +388,68 @@ UniValue getnettotals(const UniValue& params, bool fHelp)
...
+        alert.pushKV("nPriority", iter_alert.second.nPriority);
+
+        alert.pushKV("strComment", iter_alert.second.strComment);
+        alert.pushKV("strComment", iter_alert.second.strStatusBar);
+        alert.pushKV("strComment", iter_alert.second.strReserved);
+
+        alert.pushKV("IsNull()", iter_alert.second.IsNull());
+        alert.pushKV("GetHash()", iter_alert.second.GetHash().GetHex());
+        alert.pushKV("IsInEffect()", iter_alert.second.IsInEffect());
+        alert.pushKV("AppliesToMe()", iter_alert.second.AppliesToMe());
```
Comment (div72) [inline src/rpc/net.cpp]:
```suggestion
        alert.pushKV("version", iter_alert.second.nVersion);
        alert.pushKV("relay_until", DateTimeStrFormat(iter_alert.second.nRelayUntil));
        alert.pushKV("expiration", DateTimeStrFormat(iter_alert.second.nExpiration));
        alert.pushKV("id", iter_alert.second.nID);
        alert.pushKV("cancel_upto", iter_alert.second.nCancel);

        UniValue set_cancel(UniValue::VARR);
        for (const auto& cancel : iter_alert.second.setCancel) {
            set_cancel.push_back(cancel);
        }
        alert.pushKV("cancels", set_cancel);

        alert.pushKV("minimum_version", iter_alert.second.nMinVer);
        alert.pushKV("maximum_version", iter_alert.second.nMaxVer);

        UniValue set_subver(UniValue::VARR);
        for (const auto& subver : iter_alert.second.setSubVer) {
            set_subver.push_back(subver);
        }
        alert.pushKV("subversions", set_subver);

        alert.pushKV("priority", iter_alert.second.nPriority);

        alert.pushKV("comment", iter_alert.second.strComment);
        alert.pushKV("status_bar", iter_alert.second.strStatusBar);
        alert.pushKV("reserved", iter_alert.second.strReserved);

        alert.pushKV("hash", iter_alert.second.GetHash().GetHex());
        alert.pushKV("in_effect", iter_alert.second.IsInEffect());
        alert.pushKV("applies_to_me", iter_alert.second.AppliesToMe());
```
Fixed the keys, removed the IsNull and properly added cancels/subversions.

**Review by div72 — APPROVED** [review]
utACK, CI failure unrelated.

File: src/rpc/net.cpp (hunk 1 above)
Comment (jamescowens) [reply src/rpc/net.cpp]: Good.

**Comment by iFoggz** [comment]
Glad to see the keys are finally changed and reasonable height for hard fork. Looks good Jim!.

**Pull request merged**